recbole/model/general_recommender/nncf.py

```diff
@@ -0,0 +1,303 @@
+# -*- coding: utf-8 -*-
+# @Time   : 2021/1/14
+# @Author : Chengyuan Li
+# @Email  : 2017202049@ruc.edu.cn
+
+import torch
+import torch.nn as nn
+from torch.nn.init import normal_
+
+from recbole.model.abstract_recommender import GeneralRecommender
+from recbole.model.layers import MLPLayers
+from recbole.utils import InputType
+
+import numpy as np
+import networkx as nx
+import community
+from sklearn.metrics import jaccard_score
+
+
+class NNCF(GeneralRecommender):
+
+    input_type = InputType.POINTWISE
+
+    def __init__(self, config, dataset):
+        super(NNCF, self).__init__(config, dataset)
+
+        # load dataset info
+        self.LABEL = config['LABEL_FIELD']
+        self.interaction_matrix = dataset.inter_matrix(form='coo').astype(
+            np.float32)
+
+        # load parameters info
+        self.ui_embedding_size = config['ui_embedding_size']
+        self.neigh_embedding_size = config['neigh_embedding_size']
+        self.num_conv_kernel = config['num_conv_kernel']
+        self.conv_kernel_size = config['conv_kernel_size']
+        self.pool_kernel_size = config['pool_kernel_size']
+        self.mlp_hidden_size = config['mlp_hidden_size']
+        self.neigh_num = config['neigh_num']
+        self.use_random = config['use_random']
+        self.use_knn = config['use_knn']
+        self.use_louvain = config['use_louvain']

```

Comment:
How to set the neighbor data is confusing. 
It's better to replace `use_random`, `use_knn` and `use_louvain` with one parameter, which ranges in `['random', 'knn', 'louvain']`.